pkg/proc/core/linux_core.go

```diff
@@ -61,46 +73,90 @@ func readLinuxAMD64Core(corePath, exePath string) (*Process, error) {
 	if exeELF.Type != elf.ET_EXEC && exeELF.Type != elf.ET_DYN {
 		return nil, fmt.Errorf("%v is not an exe file", exeELF)
 	}
+	switch getMachineType := exeELF.Machine; getMachineType {
+	case EM_X86_64:
+		notes, err := readNotesAMD64(coreFile)
+		if err != nil {
+			return nil, err
+		}
+		memory := buildMemory(coreFile, exeELF, exe, notes)
+		entryPoint := findEntryPoint(notes)
+
+		p := &Process{
+			mem:         memory,
+			Threads:     map[int]*Thread{},
+			entryPoint:  entryPoint,
+			bi:          proc.NewBinaryInfo("linux", "amd64"),
+			breakpoints: proc.NewBreakpointMap(),
+		}
 
-	notes, err := readNotes(coreFile)
-	if err != nil {
-		return nil, err
-	}
-	memory := buildMemory(coreFile, exeELF, exe, notes)
-	entryPoint := findEntryPoint(notes)
-
-	p := &Process{
-		mem:         memory,
-		Threads:     map[int]*Thread{},
-		entryPoint:  entryPoint,
-		bi:          proc.NewBinaryInfo("linux", "amd64"),
-		breakpoints: proc.NewBreakpointMap(),
-	}
-
-	var lastThread *linuxAMD64Thread
-	for _, note := range notes {
-		switch note.Type {
-		case elf.NT_PRSTATUS:
-			t := note.Desc.(*LinuxPrStatus)
-			lastThread = &linuxAMD64Thread{linutil.AMD64Registers{Regs: &t.Reg}, t}
-			p.Threads[int(t.Pid)] = &Thread{lastThread, p, proc.CommonThread{}}
-			if p.currentThread == nil {
-				p.currentThread = p.Threads[int(t.Pid)]
+		var lastThread *linuxAMD64Thread
```

Comment:
split this code to a different function called `linuxThreadsFromNotesAMD64(*Process, []*Note)`.